nrf_rpc: Add initial protocol specification (#1312)
nrfconnect/sdk-nrfxlib: nrf_rpc/README.rst

Remote procedure call library (nRF RPC)
#######################################

nRF RPC is a remote procedure call library for enabling inter-processor communication on Nordic Semiconductor SoCs.
The library provides a function call serialization framework.

> It is RTOS-agnostic and designed to be used with an underlying transport layer, for example based on `OpenAMP`_.

Reviewer: ```suggestion
It is RTOS-agnostic and designed to be used with an underlying transport layer, such as one based on `OpenAMP`_.
```

The nRF RPC library allows for calling a function on a remote processor from a local processor, in both synchronous and asynchronous way.
It simplifies the serialization of user APIs, such as of a Bluetooth stack, and executing functions implementing those APIs on a remote processor.

Depending on the transport layer, the local and remote processors are not confined to just one device.
The remote processor can be located on a separate device of any kind, such as a PC, or it can be another CPU core within the same device.

> The library is operating system independent so it can be used with any operating system after porting the OS-dependent layers of the library.

Reviewer: ```suggestion
The library operates system independent, meaning it can be used with any operating system after porting the OS-dependent layers of the library.
```

Author: This meant that the library is independent of an operating system, but if maybe we can make it more concise and just tell that "The library can be used with any operating system..."

Reviewer: Yes, that would be clearer.
